bin/init.ml

```diff
@@ -0,0 +1,100 @@
+open Stdune
+open Import
+
+open Dune.Dune_init
+
+let doc = "Initialize dune components"
+let man =
+  [ `S "DESCRIPTION"
+  ; `P {|$(b,dune init {lib,exe,test} NAME [PATH]) initialize a new dune
+         component named $(b,NAME) from the default template for the
+         specified component kind.|}
+  ; `P {|If the optional $(b,PATH) is provided, the project will be created
+         there. Otherwise, it is created in a directory $(b,NAME) relative
+         to the current working directory.|}
+  ]
+
+let info = Term.info "init" ~doc ~man
+
+let term =
+  let%map common_term = Common.term
+  and kind_cmd =
+    (* TODO(shonfeder): Replace with nested subcommand once we have support for that *)
+    Arg.(required & pos 0 (some string) None & info [] ~docv:"INIT_KIND")
+  and name =
+    Arg.(required & pos 1 (some string) None & info [] ~docv:"NAME")
+  and path =
+    Arg.(value & pos 2 (some string) None & info [] ~docv:"PATH" )
+  and dir =
+    Arg.(value
+         & opt (some string) None
+         & info ["d"; "dir"]
+             ~docv:"DIR_NAME"
+             ~doc:"Set the name of the directory created for the \
+                   component. Set to $(b,.) to avoid creating a new \
+                   directory for the component. If omitted, a suitable \
+                   default is created")
+  and libraries =
+    Arg.(value
+         & opt (list string) []
+         & info ["libs"]
+             ~docv:"LIBRARIES"
+             ~doc:"Libraries on which the component depends")
+  and pps =
+    Arg.(value
+         & opt (list string) []
+         & info ["ppx"]
+             ~docv:"PREPROCESSORS"
+             ~doc:"ppx preprocessors used by the component")
+  and public =
+    (* TODO Move to subcommands {lib, exe} once implemented *)
+    Arg.(value
+         & flag
+         & info ["public"]
+             ~docv:"MAKE_PUBLIC"
+             ~doc:"Whether to make the component public")
+  and inline_tests =
+    (* TODO Move to subcommand lib once implemented *)
+    Arg.(value
+         & flag
+         & info ["inline-tests"]
+             ~docv:"USE_INLINE_TESTS"
+             ~doc:"Whether to use inline tests")
+  in
+  Common.set_common common_term ~targets:[];
+  match Kind.of_cmd kind_cmd with
+  | Some kind ->
+    validate_component_name name;
+
+    let open Component in
+    let context = Init_context.make common_term.root path in
+    let common = Options.{name; libraries; pps; dir} in
```

Comment:
I believe you just want `{Option. name; libraries, pps; dir}`. This will avoid triggering the unused open warning.

Another option that we use in dune is:

```
let common : Options.common = {name; libraries; pps; dir}
```